apitools/org.eclipse.pde.api.tools/src/org/eclipse/pde/api/tools/internal/model/ApiModelFactory.java

```diff
@@ -115,37 +114,19 @@ public static IApiComponent newApiComponent(IApiBaseline baseline, IPluginModelB
 		if (location == null) {
 			return null;
 		}
-		BundleComponent component = null;
-		if (isBinaryProject(location)) {
-			component = new BundleComponent(baseline, location, getBundleID());
-		} else {
+		BundleComponent component;
+		if (model.getUnderlyingResource() instanceof IProject project && project.exists()
```

Comment:
is the underlying resource possibly not a project but  particular file?

Reply:
Yes it is indeed the MANIFEST.MF.